packages/create-plugin/src/migrations/utils.ts

```diff
@@ -48,3 +51,83 @@ export function flushChanges(context: Context) {
 }
 
 export const migrationsDebug = debug.extend('migrations');
+
+/**
+ * Formats the files in the migration context using the version of prettier found in the local node_modules.
+ * If prettier isn't installed or the file is ignored or has no parser, it will not be formatted.
+ *
+ * @param context - The context to format.
+ */
+export async function formatFiles(context: Context) {
+  let prettier;
+  const require = createRequire(import.meta.url);
+  // import.meta.resolve parent arg doesn't change base path for bare specifiers so need to use require.resolve
+  const localPrettierPath = require.resolve('prettier', { paths: [process.cwd()] });
+
+  try {
+    prettier = await import(localPrettierPath);
+  } catch (error) {
+    // don't do anything if prettier is not installed
+  }
+
+  if (!prettier) {
+    return;
+  }
+
+  if (prettier.default) {
+    prettier = prettier.default;
+  }
+
+  const files = context.listChanges();
+  for (const [filePath, { content, changeType }] of Object.entries(files)) {
+    if (changeType !== 'delete' && content) {
+      const prettierOptions = await prettier.resolveConfig(filePath);
+      const supported = await prettier.getFileInfo(filePath, prettierOptions as any);
+
+      if (filePath.endsWith('.eslintrc')) {
+        supported.inferredParser = 'json';
+      }
+      if (supported.ignored || !supported.inferredParser) {
+        continue;
+      }
+
+      files[filePath] = {
+        ...files[filePath],
+        content: await prettier.format(content, {
+          ...prettierOptions,
+          parser: supported.inferredParser,
+        }),
+      };
+    }
+  }
+}
+
+// Cache the package.json contents to avoid re-installing dependencies if the package.json hasn't changed
```

Comment:
Nit: maybe adding some info to the "why" can be useful.

```suggestion
// Cache the package.json contents to avoid re-installing dependencies if the package.json hasn't changed
// (This runs for each migration used in an update)
```